src/components/RichTextEditor/BlurObserver.ts

```diff
@@ -0,0 +1,42 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { floatingLinkActions, useEditorRef, useFloatingLinkSelectors } from '@udecode/plate';
+import { useCallback, useEffect } from 'react';
+
+export const BlurObserver = () => {
+    const editorRef = useEditorRef();
+    const isFloatingModalOpen = useFloatingLinkSelectors().isOpen(editorRef.id);
+
+    const collapseEverything = useCallback(() => {
+        editorRef.collapse();
+        if (isFloatingModalOpen) {
+            floatingLinkActions.hide();
+        }
+    }, [editorRef, isFloatingModalOpen]);
+
+    useEffect(() => {
+        const handleInteractWithOutside = (event: FocusEvent | MouseEvent) => {
+            const richTextEditorSelector = `[data-editor-id='${editorRef.id}']`;
+            const toolbarSelector = "[data-selector='toolbar-floating']";
+            const richTextEditorWrapper = document.querySelector<HTMLDivElement>(richTextEditorSelector);
```

Comment:
What do you think about using a `ref` for the `richTextEditorWrapper` and `toolbarWrapper`? So we just have to query the DOM once

Reply:
Hmmm.. toolbar does not always exist, and same goes for modals, so we can't really just query them once. We could do it for the RTE wrapper, but at that point I would keep it consistent. What do you think?

Reply:
True, didn't think about that its completely removed from the DOM